crates/services/consensus_module/poa/src/service.rs

```diff
@@ -387,6 +402,70 @@ where
         Ok(())
     }
 
+    async fn produce_predefined_block(
+        &mut self,
+        predefined_block: &Block,
+        chain_id: &ChainId,
+    ) -> anyhow::Result<()> {
+        tracing::info!("Producing predefined block");
+        let last_block_created = Instant::now();
+        // verify signing key is set
+        if self.signing_key.is_none() {
+            return Err(anyhow!("unable to produce blocks without a consensus key"))
+        }
+
+        // Ask the block producer to create the block
+        let (
+            ExecutionResult {
+                block,
+                skipped_transactions,
+                tx_status,
+                events,
+            },
+            changes,
+        ) = self
+            .block_producer
+            .produce_predefined_block(predefined_block)
+            .await?
+            .into();
+
+        if !skipped_transactions.is_empty() {
+            tracing::error!("During block production got invalid transactions");
+            let txs = predefined_block.transactions();
+
+            for (tx_id, err) in skipped_transactions {
+                let maybe_tx = txs.iter().find(|tx| tx.id(chain_id) == tx_id);
+                if let Some(tx) = maybe_tx {
+                    tracing::error!(
+                "During block production got invalid transaction {:?} with error {:?}",
```

Comment:
We agreed that in the case of discarded predefined transactions we want to print them in JSON format with some `BEGIN {JSON_TX} END` marker.